### PR TITLE
add neo-test projects

### PR DESCRIPTION
**Pull request by Ashuaidehao**
Relate https://github.com/neo-project/neo-express/issues/287.

### CODE REVIEW

**Comment by Liaojinghui** [comment]
@cschuchardt88

File: test/build-tasks-sample/.config/dotnet-tools.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "version": 1,
+  "isRoot": true,
+  "tools": {
+    "neo.express": {
+      "version": "3.5.19",
```
Comment (cschuchardt88) [inline test/build-tasks-sample/.config/dotnet-tools.json:6]: can you update to new versions? Same for one below.

File: test/build-tasks-sample/build-tasks-sample.sln [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+		{524B2C6C-4354-42E7-A911-F55D877D7F0E}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{524B2C6C-4354-42E7-A911-F55D877D7F0E}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{524B2C6C-4354-42E7-A911-F55D877D7F0E}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{524B2C6C-4354-42E7-A911-F55D877D7F0E}.Release|Any CPU.Build.0 = Release|Any CPU
+		{1E87FB71-E3AD-444C-884D-E511758F9E9F}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{1E87FB71-E3AD-444C-884D-E511758F9E9F}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{1E87FB71-E3AD-444C-884D-E511758F9E9F}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{1E87FB71-E3AD-444C-884D-E511758F9E9F}.Release|Any CPU.Build.0 = Release|Any CPU
+	EndGlobalSection
+EndGlobal
```
Comment (cschuchardt88) [inline test/build-tasks-sample/build-tasks-sample.sln:28]: Remove this and add to `neo-express.sln` in a new folder under `test` called `samples`?

File: test/build-tasks-sample/src/contract.csproj [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <NeoContractName>$(AssemblyName)</NeoContractName>
+    <NeoExpressBatchFile>../express.batch</NeoExpressBatchFile>
+    <Nullable>enable</Nullable>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (cschuchardt88) [inline test/build-tasks-sample/src/contract.csproj:7]: after tools is updated, can you change this to `dotnet` 7.0 or you can create a `Directory.Build.props` file in the test folder?

File: test/build-tasks-sample/test/contract-test.csproj [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <IsPackable>false</IsPackable>
+    <Nullable>enable</Nullable>
+    <RootNamespace>ContractTests</RootNamespace>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (cschuchardt88) [inline test/build-tasks-sample/test/contract-test.csproj:7]: Update to `dotnet` 7.0

**Review by cschuchardt88 — CHANGES_REQUESTED** [review]
Do we need `test/test-collector/testFiles` folder? Isn't that generated when `dotnet test`. If you don't know than we can keep it.

File: test/build-tasks-sample/.config/dotnet-tools.json (hunk 1 above)
Comment (Ashuaidehao) [reply test/build-tasks-sample/.config/dotnet-tools.json:6]: OK

**Comment by Ashuaidehao** [comment]
> Do we need `test/test-collector/testFiles` folder? Isn't that generated when `dotnet test`. If you don't know than we can keep it.

These files was contained in [origin repo](https://github.com/N3developertoolkit/neo-test/tree/main/test/test-collector/testFiles) and seems to be used in [code](https://github.com/N3developertoolkit/neo-test/blob/main/test/test-collector/UnitTest1.cs). I think we can keep it for now.

File: test/build-tasks-sample/build-tasks-sample.sln (hunk 2 above)
Comment (Ashuaidehao) [reply test/build-tasks-sample/build-tasks-sample.sln:28]: I find this directory are more like an example to show how to use the neo-test tools, even not a part of unit tests.
 Maybe we should move it the root directory with a `samples` folder? It does not make sence to be placed here.

Comment (cschuchardt88) [reply test/build-tasks-sample/build-tasks-sample.sln:28]: Sounds good, that makes more sense.

File: samples/Directory.Build.props [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+    <PropertyGroup>
+        <NeoTestVersion>3.5.17</NeoTestVersion>
```
Comment (cschuchardt88) [inline samples/Directory.Build.props:4]: Shouldn't we add reference to this project?

Comment (Ashuaidehao) [reply samples/Directory.Build.props:4]: OK

Comment (Ashuaidehao) [reply samples/Directory.Build.props]: I find this property could be used in [test.yml](https://github.com/N3developertoolkit/neo-test/blob/60b08e4328c59798d25c133c523ecf011beef5e9/.github/workflows/test.yml#L69), changing it back to unspecified value.

**Comment by cschuchardt88** [comment]
what i mean is, reference the project from the repo not from nuget

**Comment by Ashuaidehao** [comment]
> repo

It's a independent solution for using and testing the nuget package, I think it's not fit to reference other peojects directly.

**Comment by Liaojinghui** [comment]
All tests pases. I would merge it. If any issues need to be fixed. Can be fixed in another pr.

**Pull request merged**